### ISSUE TITLE
Email template with lots of HTML results in empty email (eventually)

### ISSUE THREAD

**Issue by mbassiouny33** [InvoicePlane/InvoicePlane#748]
Hi,

Recently moved from invoice Ninja to Invoice Plane and so far, nothing seem to work with ease.

One of the issues I encountered is when the email body is contained inside a div and there is a style, it shows correctly in the preview but the email is sent empty even though it appears in the preview.

Example:

From email template:
![image](https://user-images.githubusercontent.com/12175103/128600210-bb709572-194c-4f37-94ed-a6e4ebe7a62c.png)

You can see the template is fine and the preview gives what I want.

When I use this template in my invoice the body of the template is not automatically pasted, it remains empty:
![image](https://user-images.githubusercontent.com/12175103/128600266-0324f6e5-2c25-4da5-9288-8cfefd1e6e1f.png)

Please note that if I remove the style tag, it works fine, adding the style inline with <... style="..."> works.

I managed to do what I want with the inline workaround but it is not ideal.

Thanks in advance for considering my request.

**Comment by bminer** [comment]
I can also confirm this bug. If `<style>` tag is used, the email template is not copied when it's selected.

**Comment by nielsdrost7** [comment]
Can you confirm that if you copy/paste just the html that it works

Time to move this to https://community.invoiceplane.com,  it's a forum where we cab figure out which part of the hrml to copy

**Comment by bminer** [comment]
@nielsdrost7 - Any time a `<style>` tag is used, the email template is not copied to the "body" textarea when it's selected from the dropdown.

**Comment by mbassiouny33** [comment]
> Can you confirm that if you copy/paste just the html that it works

If only OP bothered to confirm this 2 years ago....😶

> Please note that if I remove the style tag, it works fine, adding the style inline with <... style="..."> works.

**Comment by nielsdrost7** [comment]
> @nielsdrost7 - Any time a `<style>` tag is used, the email template is not copied to the "body" textarea when it's selected from the dropdown.
> 
> Sorry, does that answer your question?

Could you please show steps to reproduce?
In my workload I just copy/paste html and let's go, but apparently there's a reproducible bug.
I need exact steps

**Comment by bminer** [comment]
Sure thing!

1. Login then go to Settings > Email Templates
2. Click New
3. Enter a title "Test"
4. Enter a body:
```html
<style>
p {
  font-size: 20px;
}
</style>
<p>This is a test</p>
```
5. Click Save
6. Open any invoice
7. Click Options > Send Email
8. Select Email Template dropdown > Click "Test" email template you created in steps 1-5.
9. Notice that email template is not copied into "Body"

Removing the `<style>` tag from the email template makes everything work properly again.

**Comment by bminer** [comment]
There's an easy workaround, which is to avoid `<style>` tags altogether.

For example, your HTML body could be:
```html
<p style="font-size: 20px;">This is a test</p>
```

... but you'd have to do this for each `<p>` tag.

If I had to guess, I'd say that some security feature is stripping the `<style>` tag.

**Comment by mbassiouny33** [comment]
> There's an easy workaround, which is to avoid <style> tags altogether.

you don't say...

Do you guys even read what I wrote? I kind of feel ignored and transparent like no one sees me.


I literally wrote:

> " adding the style inline with <... style="..."> works.>


Also 

> I need exact steps

The steps to reproduce were given in the first post as well!



Paste this in the body text area (as seen in the screenshot posted in the first post) in **a template** (yes the screenshot shows a template)
Open an invoice
```
<style> #myID{background-color:brown;}</style>
Some text before
<div id="myID"> some here here </div>
some text after
```
Click <send email> at the top. Baam, see the issue.

**this was the case when I opened the issue, but surprisingly I found out something very important today.**
If you use the `<style>` tag in mail body on the fly NOT in a template , like for example
https://yoursite/index.php/invoices/view/<invoide id>

Things seem to work fine.


> If I had to guess, I'd say that some security feature is detecting the <style> tag and causing the bug.
Yes, wild blind guess without opening the code, it seems there  a sanitization function somewhere that cleans **the template** before pasting it in the mail body.



Also I know some of our instances are using style tag and RTL for clients that receive rtl invoices, which means I found a workaround _2 years ago_ somehow, but ofc  I don't remember how I did it, so I am gonna need to check.

So to sum up, there are 2 behaviours :

1-  When parsing to the message body from a template, anything using CSS is stripped and the <style> tag is removed
2- Pick an invoice, click send email, edit body and put HTML, then send, it gets sent correctly and CSS shows up.


Since the issue is 2 years old, I do not remember why I said "results in empty email", it is not 100% empty, it is just stripped of styled content (like you see in screenshot 2 of my post), Maybe back then my brain made me write "empty" because I had my entire messages back then in a style div? I will check that as well.

**Comment by nielsdrost7** [comment]
thanks @bminer , those are great reproduction steps!

**Comment by nielsdrost7** [comment]
@mbassiouny33 this is not a forum where you can just say what you want.

I appreciate bminer's steps to reproduce and that's it.
Glad you found the workaround of not copying those style tags, so you can just keep using those email templates.

Next time, just open an issue on the forums, it would have been better and prevented this clutter.

**Comment by mbassiouny33** [comment]
> @mbassiouny33 this is not a forum where you can just say what you want.

> Next time, just open an issue on the forums, it would have been better and prevented this clutter.

No I won't. "Say whatever I want"? how is that relevant to anything? I am literally the one contributing the most useful details in this Github issue. 

"Clutter"?
I am sorry for polluting your paradise.
This is an *ISSUE* reported on *github issues*, with steps to reproduce and different behaviors in different cases, it is the right place. 

In the interest of remaining on-topic and discussing technical stuff, I am just gonna assume this is a language barrier and your choice of words is just really bad because there is no other logical explanation for your utterly rude and condescending choice of words.
